### PR TITLE
refactoring parameter check on clustering feature.

### LINKED ISSUE #314: Clustering: parameter check is too strict
Clustering check all parameters no matter compressor_method is "simple".

According to document, if compressor_method is "simple", bucket_length, compressed_bucket_size, bicriteria_base_size, forgetting_factor, forgetting_threshold don't check.

ref:http://jubat.us/en/api_clustering.html
"When compressor_method is specified as simple, the following parameters will be ignored: bucket_length, compressed_bucket_size, bicriteria_base_size, forgetting_factor, forgetting_threshold."

### PR DESCRIPTION
**Pull request by yukimori**
fix #314 

if compressor_method is simple, bucket_length, compressed_bucket_size,bicriteria_base_size,forgetting_factor,forgetting_threshold are ignored.

In addtion, modify bucket_size range
modify bucket_size range from 2<=bucket_size to 1<=bucket_size.

### CODE REVIEW

File: jubatus/core/clustering/storage_factory.cpp [hunk 1]
```diff
@@ -29,27 +29,42 @@ namespace clustering {
...
+      s->set_compressor(jubatus::util::lang::shared_ptr<compressor::compressor>(
+          new compressor::gmm_compressor(config)));
+      ret.reset(s);
 #endif
-  } else if (config.compressor_method == "simple") {
-    simple_storage *s = new simple_storage(name, config);
-    ret.reset(s);
+    } else {
+      throw JUBATUS_EXCEPTION(
+          common::invalid_parameter(config.compressor_method));
```
Comment (kmaehashi) [inline jubatus/core/clustering/storage_factory.cpp]: I think it's better to throw ``unsupported_method()``.
The first argument of ``invalid_parameter`` is an expected value.
https://github.com/jubatus/jubatus_core/blob/0.3.4/jubatus/core/common/exception.hpp#L372-L378

File: jubatus/core/clustering/storage_factory.cpp [hunk 2]
```diff
@@ -29,27 +29,42 @@ namespace clustering {
...
+    }
   } else {
-    throw JUBATUS_EXCEPTION(
-        common::unsupported_method(config.compressor_method));
+      throw JUBATUS_EXCEPTION(
+          common::invalid_parameter(method));
```
Comment (kmaehashi) [inline jubatus/core/clustering/storage_factory.cpp]: The same as above.

**Comment by kmaehashi** [comment]
:+1: LGTM except for comments above.

Please update the clustering API reference for ``bucket_size`` restriction changes.

**Comment by yukimori** [comment]
Thank you for your comment.
I changend "invalid_parameter" to "unsupported_method" according to your comment.

**Comment by kmaehashi** [comment]
:+1:

**Pull request merged**